cmd/ipfs/pinmfs.go

```diff
@@ -169,12 +169,16 @@ func pinMFS(
 		pinName = fmt.Sprintf("policy/%s/mfs", node.Identity().String())
 	}
 
-	// check if same pin exists
-	lsPinCh, lsErrCh := c.Ls(ctx, pinclient.PinOpts.FilterCIDs(cid), pinclient.PinOpts.FilterName(pinName))
-	pinFound := false
+	// check if MFS pin exists (across all possible states) and inspect its CID
+	pinStatuses := []pinclient.Status{pinclient.StatusQueued, pinclient.StatusPinning, pinclient.StatusPinned, pinclient.StatusFailed}
+	lsPinCh, lsErrCh := c.Ls(ctx, pinclient.PinOpts.FilterName(pinName), pinclient.PinOpts.FilterStatus(pinStatuses...))
+	existingRequestID := "" // is there any pre-existing MFS pin with pinName (for any CID)?
+	alreadyPinned := false  // is CID for current MFS already pinned?
 	for ps := range lsPinCh {
+		existingRequestID = ps.GetRequestId()
```

Comment:
IIUC this means we're replacing the oldest pin with the correct name. This doesn't seem unreasonable, but it's non-obvious so I wanted to clarify that this is the intended behavior.

Reply:
Correct. The assumption is name is unique enough to not have more than a single pre-existing one, but if for some reason there is more, we just replace last one we inspected.